src/main/java/com/medallia/word2vec/SearcherImpl.java

```diff
@@ -79,9 +72,22 @@ private double calculateDistance(double[] otherVec, double[] vec) {
 	 * @throws UnknownWordException If word is not in the model's vocabulary
 	 */
 	private double[] getVector(String word) throws UnknownWordException {
-		if (!normalized.containsKey(word))
+		final double[] result = getVectorOrNull(word);
+		if(result == null)
 			throw new UnknownWordException(word);
-		return normalized.get(word);
+
+		return result;
+	}
+
+	private double[] getVectorOrNull(final String word) {
+    final Integer index = word2vectorOffset.get(word);
+		  if(index == null)
+			  return null;
+
+		double[] result = new double[model.layerSize];
+		model.vectors.position(index);
```

Comment:
Hrm, this is inherently not thread safe.  Can we add Javadoc to the class (and the NotThreadSafe annotation if possible)?